app/src/main/java/framgia/com/ichat/data/repository/SignUpRepository.java

```diff
@@ -0,0 +1,29 @@
+package framgia.com.ichat.data.repository;
+
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.firebase.auth.FirebaseUser;
+
+import framgia.com.ichat.data.source.AuthenticationDataSource;
+
+public class SignUpRepository implements AuthenticationDataSource.Remote {
```

Comment:
sao tên datasource 1 kiểu mà repository lại 1 kiểu
```suggestion
public class AuthenticationRepository implements AuthenticationDataSource.Remote {
```